From 96a9eaa0ef58d5388cfb5766cca34a4e7026615a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADctor=20Manuel=20J=C3=A1quez=20Leal?=
 <vjaquez@igalia.com>
Date: Tue, 6 Jun 2023 13:27:39 +0200
Subject: [PATCH 129/205] msdk: remove libva check before 1.12

Since gstmsdk depends on gstva library it also bumps the required version.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4781>
---
 sys/msdk/gstmsdkallocator_libva.c | 12 ------------
 sys/msdk/gstmsdkvpp.c             |  6 ------
 sys/msdk/msdk_libva.c             |  4 +---
 3 files changed, 1 insertion(+), 21 deletions(-)

diff --git a/sys/msdk/gstmsdkallocator_libva.c b/sys/msdk/gstmsdkallocator_libva.c
index 1c046b740..b2ab8bf6d 100644
--- a/sys/msdk/gstmsdkallocator_libva.c
+++ b/sys/msdk/gstmsdkallocator_libva.c
@@ -331,7 +331,6 @@ gst_msdk_frame_lock (mfxHDL pthis, mfxMemId mid, mfxFrameData * data)
         data->Y = data->V + 2;
         data->A = data->V + 3;
         break;
-#if VA_CHECK_VERSION(1, 4, 1)
       case VA_FOURCC_A2R10G10B10:
         data->Pitch = mem_id->image.pitches[0];
         data->R = buf + mem_id->image.offsets[0];
@@ -339,8 +338,6 @@ gst_msdk_frame_lock (mfxHDL pthis, mfxMemId mid, mfxFrameData * data)
         data->B = data->R;
         data->A = data->R;
         break;
-#endif
-#if VA_CHECK_VERSION(1, 2, 0)
       case VA_FOURCC_Y210:
       case VA_FOURCC_Y216:
         data->Pitch = mem_id->image.pitches[0];
@@ -359,7 +356,6 @@ gst_msdk_frame_lock (mfxHDL pthis, mfxMemId mid, mfxFrameData * data)
         data->V = data->U + 4;
         data->A = data->U + 6;
         break;
-#endif
       case VA_FOURCC_ABGR:
         data->Pitch = mem_id->image.pitches[0];
         data->R = buf + mem_id->image.offsets[0];
@@ -517,11 +513,7 @@ gst_msdk_export_dmabuf_to_vasurface (GstMsdkContext * context,
       va_fourcc = VA_FOURCC_YUY2;
       break;
     case GST_VIDEO_FORMAT_P010_10LE:
-#if VA_CHECK_VERSION(1, 2, 0)
       va_chroma = VA_RT_FORMAT_YUV420_10;
-#else
-      va_chroma = VA_RT_FORMAT_YUV420_10BPP;
-#endif
       va_fourcc = VA_FOURCC_P010;
       break;
     case GST_VIDEO_FORMAT_UYVY:
@@ -538,13 +530,10 @@ gst_msdk_export_dmabuf_to_vasurface (GstMsdkContext * context,
       va_chroma = VA_RT_FORMAT_YUV444;
       va_fourcc = VA_FOURCC_AYUV;
       break;
-#if VA_CHECK_VERSION(1, 4, 1)
     case GST_VIDEO_FORMAT_BGR10A2_LE:
       va_chroma = VA_RT_FORMAT_RGB32_10;
       va_fourcc = VA_FOURCC_A2R10G10B10;
       break;
-#endif
-#if VA_CHECK_VERSION(1, 2, 0)
     case GST_VIDEO_FORMAT_Y210:
       va_chroma = VA_RT_FORMAT_YUV422_10;
       va_fourcc = VA_FOURCC_Y210;
@@ -565,7 +554,6 @@ gst_msdk_export_dmabuf_to_vasurface (GstMsdkContext * context,
       va_chroma = VA_RT_FORMAT_YUV444_12;
       va_fourcc = VA_FOURCC_Y416;
       break;
-#endif
 #if (MFX_VERSION >= 2004)
     case GST_VIDEO_FORMAT_RGBP:
       va_chroma = VA_RT_FORMAT_RGBP;
diff --git a/sys/msdk/gstmsdkvpp.c b/sys/msdk/gstmsdkvpp.c
index df27fe538..6022d3aa1 100644
--- a/sys/msdk/gstmsdkvpp.c
+++ b/sys/msdk/gstmsdkvpp.c
@@ -59,15 +59,9 @@
 #include "gstmsdkvpputil.h"
 #include "gstmsdkallocator.h"
 
-#define EXT_FORMATS     ""
-
 #ifndef _WIN32
 #include "gstmsdkallocator_libva.h"
 #include <gst/va/gstvaallocator.h>
-#if VA_CHECK_VERSION(1, 4, 1)
-#undef EXT_FORMATS
-#define EXT_FORMATS     ", BGR10A2_LE"
-#endif
 #else
 #include <gst/d3d11/gstd3d11.h>
 #endif
diff --git a/sys/msdk/msdk_libva.c b/sys/msdk/msdk_libva.c
index 8ebc3e121..140e27773 100644
--- a/sys/msdk/msdk_libva.c
+++ b/sys/msdk/msdk_libva.c
@@ -71,10 +71,8 @@ static const struct fourcc_map gst_msdk_fourcc_mfx_to_va[] = {
   FOURCC_MFX_TO_VA (RGB565, RGB565),
 #endif
   FOURCC_MFX_TO_VA (AYUV, AYUV),
-#if VA_CHECK_VERSION(1, 4, 1)
   FOURCC_MFX_TO_VA (A2RGB10, A2R10G10B10),
-#endif
-#if ((MFX_VERSION >= 1027) && VA_CHECK_VERSION(1, 2, 0))
+#if (MFX_VERSION >= 1027)
   FOURCC_MFX_TO_VA (Y210, Y210),
   FOURCC_MFX_TO_VA (Y410, Y410),
 #endif
-- 
2.40.1

